docs/source/release_notes/2024.1.rst

```diff
@@ -0,0 +1,118 @@
+pyMOR 2024.1 (July 8, 2025)
+-------------------------------
+
+We are proud to announce the release of pyMOR 2024.1!
+
+The main new features are:
+
+* vector fitting
+
+* successive constraints method
+
+* shifted Cholesky QR algorithm
+
+* improved randomized range finder
+
+* additional tutorials and MOR methods overview
```

Comment:
We never had this list before but instead stated how many commits entered the release. I am fine with both versions actually.